// @node-minify/html-minifier/lib/html-minifier.js
"use strict";

var _htmlMinifier = _interopRequireDefault(require("html-minifier"));

var _utils = require("@node-minify/utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/*!
 * node-minify
 * Copyright(c) 2011-2020 Rodolphe Stoclin
 * MIT Licensed
 */

/**
 * Module dependencies.
 */

/**
 * Module variables.
 */
const HTMLMinifier = _htmlMinifier.default.minify;
const defaultOptions = {
  collapseBooleanAttributes: true,
  collapseInlineTagWhitespace: true,
  collapseWhitespace: true,
  minifyCSS: true,
  minifyJS: true,
  removeAttributeQuotes: true,
  removeCDATASectionsFromCDATA: true,
  removeComments: true,
  removeCommentsFromCDATA: true,
  removeEmptyAttributes: true,
  removeOptionalTags: true,
  removeRedundantAttributes: true,
  removeScriptTypeAttributes: true,
  removeStyleLinkTypeAttributes: true,
  useShortDoctype: true
};
/**
 * Run html-minifier.
 *
 * @param {Object} settings
 * @param {String} content
 * @param {Function} callback
 */

const minifyHTMLMinifier = ({
  settings,
  content,
  callback,
  index
}) => {
  const options = Object.assign({}, defaultOptions, settings.options);
  const contentMinified = HTMLMinifier(content, options);

  if (!settings.content) {
    _utils.utils.writeFile({
      file: settings.output,
      content: contentMinified,
      index
    });
  }

  if (callback) {
    return callback(null, contentMinified);
  }

  return contentMinified;
};
/**
 * Expose `minifyHTMLMinifier()`.
 */


module.exports = minifyHTMLMinifier;
